Mathlib/SetTheory/Game/PGame.lean

```diff
@@ -732,58 +732,40 @@ theorem leftResponse_spec {x : PGame} (h : 0 ≤ x) (j : x.RightMoves) :
-    upperBound (Subtype.val ∘ (equivShrink s).symm) ∈ upperBounds s := fun i hi ↦ by
-  simpa using le_upperBound (Subtype.val ∘ (equivShrink s).symm) (equivShrink s ⟨i, hi⟩)
-#align pgame.upper_bound_mem_upper_bounds SetTheory.PGame.upperBound_mem_upperBounds
-
-/-- A small set `s` of pre-games is bounded above. -/
-lemma bddAbove_of_small (s : Set PGame.{u}) [Small.{u} s] : BddAbove s :=
-  ⟨_, upperBound_mem_upperBounds s⟩
+#noalign pgame.upper_bound
+#noalign pgame.upper_bound_right_moves_empty
+#noalign pgame.le_upper_bound
+#noalign pgame.upper_bound_mem_upper_bounds
+
+/-- A small family of pre-games is bounded above. -/
+lemma bddAbove_range_of_small [Small.{u} ι] (f : ι → PGame.{u}) : BddAbove (Set.range f) := by
+  let x : PGame.{u} := ⟨Σ i, (f $ (equivShrink.{u} ι).symm i).LeftMoves, PEmpty,
+    fun x ↦ moveLeft _ x.2, PEmpty.elim⟩
+  refine ⟨x, Set.forall_range_iff.2 fun i ↦ ?_⟩
+  rw [← (equivShrink ι).symm_apply_apply i, le_iff_forall_lf]
+  simpa using fun j ↦ @moveLeft_lf x ⟨equivShrink ι i, j⟩
+
+/-- A small set of pre-games is bounded above. -/
+lemma bddAbove_of_small (s : Set PGame.{u}) [Small.{u} s] : BddAbove s := by
+  simpa using bddAbove_range_of_small (Subtype.val : s → PGame.{u})
 #align pgame.bdd_above_of_small SetTheory.PGame.bddAbove_of_small
 
-/-- An explicit lower bound for a family of pre-games, whose right moves are the union of the right
-moves of all the pre-games in the family. -/
-def lowerBound {ι : Type u} (f : ι → PGame.{u}) : PGame :=
-  ⟨PEmpty, Σ i, (f i).RightMoves, PEmpty.elim, fun x => moveRight _ x.2⟩
-#align pgame.lower_bound SetTheory.PGame.lowerBound
-
-instance lowerBound_leftMoves_empty {ι : Type u} (f : ι → PGame.{u}) :
-    IsEmpty (lowerBound f).LeftMoves := instIsEmptyPEmpty
-#align pgame.lower_bound_left_moves_empty SetTheory.PGame.lowerBound_leftMoves_empty
-
-lemma lowerBound_le {ι : Type u} (f : ι → PGame.{u}) (i : ι) : lowerBound f ≤ f i := by
-  rw [lowerBound, le_iff_forall_lf]
-  dsimp
-  simp only [IsEmpty.forall_iff, true_and_iff]
-  exact fun j ↦ @lf_moveRight (lowerBound f) ⟨i, j⟩
-#align pgame.lower_bound_le SetTheory.PGame.lowerBound_le
-
-lemma lowerBound_mem_lowerBounds (s : Set PGame.{u}) [Small.{u} s] :
-    lowerBound (Subtype.val ∘ (equivShrink s).symm) ∈ lowerBounds s := fun i hi => by
-  simpa using lowerBound_le (Subtype.val ∘ (equivShrink s).symm) (equivShrink s ⟨i, hi⟩)
-#align pgame.lower_bound_mem_lower_bounds SetTheory.PGame.lowerBound_mem_lowerBounds
-
-/-- A small set `s` of pre-games is bounded below. -/
-lemma bddBelow_of_small (s : Set PGame.{u}) [Small.{u} s] : BddBelow s :=
-  ⟨_, lowerBound_mem_lowerBounds s⟩
+#noalign pgame.lower_bound
+#noalign pgame.lower_bound_left_moves_empty
+#noalign pgame.lower_bound_le
+#noalign pgame.lower_bound_mem_lower_bounds
+
+/-- A small family of pre-games is bounded below. -/
+lemma bddBelow_range_of_small [Small.{u} ι] (f : ι → PGame.{u}) : BddBelow (Set.range f) := by
```

Comment:
Same question

Reply:
No need to point out everything twice. Despite contrary claims, I am in fact not a LLM 😉